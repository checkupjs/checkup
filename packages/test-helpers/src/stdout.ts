import { stdout as pso } from 'stdout-stderr';

beforeEach(() => {
  pso.start();
});

afterEach(() => {
  pso.stop();
});

/**
 * Uses jest's beforeEach/afterEach to setup/teardown capturing
 * process.stdout in order to use for assertions.
 *
 * @export
 * @returns
 */
export function stdout() {
  return pso.output;
}
