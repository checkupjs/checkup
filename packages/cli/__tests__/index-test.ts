import { stdout } from './__utils__/stdout';

import cmd = require('../src');

describe('@checkup/cli', () => {
  it('should output checkup result', async () => {
    await cmd.run([]);

    expect(stdout()).toMatchSnapshot();
  });
});
