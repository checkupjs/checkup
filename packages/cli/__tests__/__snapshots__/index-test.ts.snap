// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@checkup/cli should output checkup result 1`] = `
"=== Project Information

name:    @checkup/cli
type:    unknown
version: 0.0.0

=== Types

Type                  Total 
components            0     
controllers           0     
helpers               0     
initializers          0     
instance-initializers 0     
mixins                0     
models                0     
routes                0     
services              0     
templates             0     

=== Test Types

"
`;
